yt/fields/field_info_container.py

```diff
@@ -463,14 +469,47 @@ def keys(self):
         return keys
 
     def check_derived_fields(self, fields_to_check=None):
+        # define
+        # - a blacklist (exceptions that we know should be caught)
+        # - a whitelist (exceptions that should be handled)
+        # - a greylist (excpetions that may hide bugs but should be checked)
+
+        # in the long run, the greylist should be removed
+        blacklist = (RecursionError,)
+        whitelist = (NotImplementedError,)
+        greylist = (
+            YTFieldNotFound,
+            YTDomainOverflow,
+            YTCoordinateNotImplemented,
+            NeedsConfiguration,
+            TypeError,
+            ValueError,
+            IndexError,
+            AttributeError,
+            KeyError,
+            # code smells -> those are very likely bugs
+            UnitConversionError,  # solved in GH PR 2897 ?
+        )
+
         deps = {}
         unavailable = []
         fields_to_check = fields_to_check or list(self.keys())
         for field in fields_to_check:
             fi = self[field]
             try:
+                # Here we except a very large number of unrelated exceptions.
+                # This is a code smell and indicates that the single line in the try
+                # block does too much, or has an impredictible behaviour.
+                # Each exception caught is likely covering a bug somewhere in the test
+                # suite. They are sorted so it's easier to keep track of.
+                # In order to solve these, open a PR with one (and preferably only one)
+                # of those deactivated, then see which tests fail and solve the
+                # underlying issues locally.
                 fd = fi.get_dependencies(ds=self.ds)
-            except (NotImplementedError, Exception) as e:  # noqa: B014
+            except blacklist as err:
+                print(f"{err.__class__} raised for field {field}")
```

Comment:
is there a reason for using `print` here instead of `mylog`?

Reply:
+1 to this.